### PR TITLE
Improve http-kit integration / test coverage

### PR DESCRIPTION
**Pull request by vemv**
See also: https://github.com/clojure-emacs/refactor-nrepl/issues/288

### CODE REVIEW

File: src/refactor_nrepl/artifacts.clj [hunk 1]
```diff
@@ -51,35 +51,42 @@
...
   "Returns a vector of [[some/lib \"0.1\"]...]."
   []
   (try
-    (->> "https://clojars.org/repo/all-jars.clj.gz"
-         io/input-stream
-         GZIPInputStream.
-         io/reader
-         line-seq
-         (keep edn-read-or-nil))
+    (or @clojars-artifacts-cache
```
Comment (expez) [inline src/refactor_nrepl/artifacts.clj:60]: I believe this breaks the `force` functionality in `artifact-list` which forces an update of a potentially stale cache.

**Review by expez — COMMENTED** [review]
Looks good otherwise :)

File: src/refactor_nrepl/artifacts.clj (hunk 1 above)
Comment (vemv) [reply src/refactor_nrepl/artifacts.clj:60]: Mmm good once, I briefly considered the daily caching but it didn't seem too relevant.

It is for long-lived sessions which are very much emacsy :)

**Comment by vemv** [comment]
Removed the atom cache I had added and made that one defn private instead. That way one can only use said defn via its caching façade.

**Review by expez — APPROVED** [review]
LGTM 👍

**Pull request merged**